### DISCUSSION TITLE
Delay  in airflow Schedule performance  / task executor performance

### DISCUSSION THREAD

**Discussion by DLTKHemanthKodandarama** [apache/airflow#27154] [Q&A]
Hello Everyone,

I am facing the following performance issues

Task either stuck in queued state for few secs / minutes or waiting for scheduler with no status update
Once task is set to running i can see there is delay (from 10+ secs to minutes) in context switching between executor/task_runner and worker
I am using airflow 2.3.4 version on AWS EKS kubernetes using airflow official helm chart https://airflow.apache.org/docs/helm-chart/stable/parameters-ref.html with worker and scheduler pods , aws RDS postgres database instance , AWS elasticache redis cluster(celery broker), with below airflow configuration.

1 Parallelism ------------------  3200
2. max_active_runs_per_dag--- 200
3. dag_concurrency-----------------  1000
4. worker_concurrency ----------16
5. Scheduler / triggers replicas - 4
There are no resource contraints for memory /cpu constraints on worker / scheduler pods in kubernetes, redis , db etc

[airflow_config.yml](https://i.stack.imgur.com/lNNOV.png)

[task_log_with_exector_dealy_60secs](https://i.stack.imgur.com/q8VfK.png)
[task queued](https://i.stack.imgur.com/HPJFv.png)

**Comment by o-nikolas** [comment]
Many reasons can cause scheduling delays with Airflow, and it's hard to pinpoint from the outside. It can be anything from the way your dag files are written (excessive computation or network/db requests at dag parse time), slow file system/network, if you have many thousands of concurrent tasks, etc.
There is a great write-up here in our docs regarding fine tuning the scheduler. Please do read the whole thing top to bottom to get some ideas for optimizing your environment: https://airflow.apache.org/docs/apache-airflow/stable/concepts/scheduler.html?highlight=scheduler#fine-tuning-your-scheduler-performance
